swan-lake/learn/mocking.md

```diff
@@ -0,0 +1,338 @@
+---
+layout: ballerina-left-nav-pages-swanlake
+title: Mocking
+description: Learn how to use Ballerina's built-in mocking API provided by the test to test modules
+ independent from other modules and external endpoints.
+keywords: ballerina, programming language, testing, mocking
+permalink: /swan-lake/learn/mocking/
+active: mocking
+redirect_from:
+  - /swan-lake/learn/mocking
+  - /swan-lake/learn/mocking/
+---
+
+# Mocking
+
+Mocking is useful to control the behavior of functions and objects to control the communication with other modules and external endpoints. A mock can be created by defining return values or replacing the entire object or function with a user-defined equivalent. This feature will help you to test the Ballerina code independently from other modules and external endpoints.
+
+
+## Mocking Objects
+
+The test module provides capabilities to mock an object for unit testing. This allows you to control the behavior of object member functions and values of member fields via stubbing or replacing the entire object with a user-defined equivalent. This feature will help you to test the Ballerina code independently from other modules and external endpoints.
+
+Mocking of objects can be done in two ways.
+
+
+
+1. Creating a test double (providing an equivalent object in place of the real)
```

Comment:
```suggestion
1. Creating double tests (providing an equivalent object in place of the real)
```

Reply:
"test double" is a jargon. So we can keep it as it is.